src/test/run-make/thumb-none-qemu/Makefile

```diff
@@ -0,0 +1,42 @@
+-include ../../run-make-fulldeps/tools.mk
+
+# How to run this
+# $ ./x.py clean
+# $ ./x.py test --target thumbv7m-none-eabi src/test/run-make
+
+ifeq ($(TARGET),thumbv7m-none-eabi))
+
+# For cargo setting
+RUSTC := $(RUSTC_ORIGINAL)
+LD_LIBRARY_PATH := $(HOST_RPATH_DIR)
+# We need to be outside of 'src' dir in order to run cargo
+WORK_DIR := $(TMPDIR)
+HERE := $(shell pwd)
+
+
+# hint: we could set variables per $(TARGET) basis in order to support other targets.
```

Comment:
`qemu-system-arm` only supports full system emulation of the LM3S6965 and LM3S811 microcontrollers. They are both Cortex-M3 microcontrollers so we can test the `thumbv6m-none-eabi` and `thumbv7m-none-eabi` targets with either QEMU target. The `thumbv7em-none-eabi` and `thumbv7em-none-eabihf` targets have incompatible ISAs and they can crash QEMU.

It would be great to also test the `thumbv6m-none-eabi` target here since it's an older ARM architecture. If you are up to it you can do that in a follow-up PR.

Reply:
Ahem, older... ;) Cortex-M0 is still very relevant, is there any support for a CM0 MCU in qemu?

Reply:
@therealprof I've run 4 binaries on `qemu-system-arm`. As I remember, some worked in different board setting.  My theory is if an ISA (`thumbv6m-none-eabi`) is roughly a subset of another (`thumbv7m-none-eabi`), and there's no harmful register changes, it has chance to run successfully.  I'll try with existing qemu setting for `thumbv6m` (Cortex-M0) also.

Reply:
@sekineh `thumbv6m-none-eabi` is indeed a subset of `thumbv7m-none-eabi` and code might run just fine on higher ISA versions. However to be on the safe(r) side you really want to test on a Cortex-M0(+) target or emulation.

Reply:
@therealprof these are the Cortex CPUs QEMU 3.0.0 supports:

```
  cortex-a15
  cortex-a7
  cortex-a8
  cortex-a9
  cortex-m3
  cortex-m33
  cortex-m4
  cortex-r5
  cortex-r5f
```

No `cortex-m0` or `cortex-m4f` but you can run code compiled for `thumbv6m-none-eabi` using `-cpu cortex-m3` since the M0 ISA is a subset of the M3 ISA. We can't run code compiled for the `thumbv7em-none-eabihf` though because the FPU instructions crash QEMU.

I'm not entirely sure if `-machine` is completely orthogonal to `-cpu` but I would prefer not to mix a Cortex-M3 `machine` with `-cpu cortex-m4`.

Reply:
I successfully ran the `thumv6m-none-eabi` binary on qemu `-cpu cortex-m3`
```
sekineh@sekineh-VirtualBox:~/cortex-m-rt_me$ cargo run --example qemu --target thumbv6m-none-eabi
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s                                                    
     Running `qemu-system-arm -cpu cortex-m3 -machine lm3s6965evb -nographic -semihosting-config enable=on,target=native -kernel target/thumbv6m-none-eabi/debug/examples/qemu`
x = 42
```